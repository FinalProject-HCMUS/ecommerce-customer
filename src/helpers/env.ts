export const VITE_REDIRECT_URI = import.meta.env.VITE_REDIRECT_URI as string;
export const VITE_AUTH_URI = import.meta.env.VITE_AUTH_URI as string;
export const VITE_CLIENT_ID = import.meta.env.VITE_CLIENT_ID as string;
export const VITE_BACKEND_URL = import.meta.env.VITE_BACKEND_URL as string;
export const VITE_HUGGINGFACE_API_KEY =  import.meta.env.VITE_HUGGINGFACE_API_KEY || '';
export const SERVICE_ID = import.meta.env.VITE_SERVICE_ID;
export const TEMPLATE_ID = import.meta.env.VITE_TEMPLATE_ID;
export const PUBLIC_KEY = import.meta.env.VITE_PUBLIC_KEY;