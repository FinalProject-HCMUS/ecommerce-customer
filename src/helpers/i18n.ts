import i18n from '../i18n';

// While all the constants modules are comprised when imported and depends on t(), we delay the real value until i18n is initialized
export const t = (text: string) => {
  return i18n ? i18n.t(text) : text;
};

export const tArray = (text: string[]) => {
  return i18n ? i18n.t(text) : text;
};

export const tUpperCase = (text: string) => {
  return i18n ? i18n.t(text).toUpperCase() : text;
};
export const tLowerCase = (text: string) => {
  return i18n ? i18n.t(text).toLowerCase() : text;
};
