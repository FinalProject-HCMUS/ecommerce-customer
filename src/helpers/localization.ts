import localizationConstants from '../constants/localization';
import localStorageConstants from '../constants/localStorage';

const { LOCALIZATION } = localStorageConstants;
const { REGIONS } = localizationConstants;

const getCurrentLanguage = () => {
  const language = localStorage.getItem(LOCALIZATION);
  return language || REGIONS.vi.key;
};

const changeLanguage = (language: string) => {
  if (language === getCurrentLanguage()) return;
  localStorage.setItem(LOCALIZATION, language);
  window.location.reload();
};

export { getCurrentLanguage, changeLanguage };
