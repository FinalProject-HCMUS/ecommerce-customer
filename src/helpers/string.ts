const formatCurrency = (value: number, currency: 'USD' | 'VND'): string => {
  return new Intl.NumberFormat('en-US', {
    style: 'currency',
    currency: currency,
    maximumFractionDigits: 0,
  }).format(value);
};

export { formatCurrency };
