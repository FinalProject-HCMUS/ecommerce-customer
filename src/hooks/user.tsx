import { useState } from 'react';
import { userApi } from '../services/apis/userApi';
import { UserResponse } from '../interfaces/user/UserResponse';
import { CreateUserRequest } from '../interfaces/user/CreateUserRequest';
import { UpdateUserRequest } from '../interfaces/user/UpdateUserRequest';
import { ChangePasswordRequest } from '../interfaces/user/ChangePasswordRequest';

export const useUser = () => {
  const [user, setUser] = useState<UserResponse | null>(null);
  const [loading, setLoading] = useState(false);

  // Fetch a single user by ID
  const fetchUserById = async (id: string): Promise<UserResponse | null> => {
    setLoading(true);
    try {
      const response = await userApi.getUserById(id);
      setUser(response.data || null);
      return response.data || null;
    } catch {
      return null;
    } finally {
      setLoading(false);
    }
  };

  // Fetch a single user by token
  const fetchUserByToken = async (): Promise<UserResponse | null> => {
    setLoading(true);
    try {
      const response = await userApi.getUserByToken();
      setUser(response.data || null);
      return response.data || null;
    } catch {
      return null;
    } finally {
      setLoading(false);
    }
  };

  // Create a new user
  const createUser = async (data: CreateUserRequest): Promise<UserResponse | null> => {
    setLoading(true);
    try {
      const response = await userApi.createUser(data);
      return response.data || null;
    } catch {
      return null;
    } finally {
      setLoading(false);
    }
  };

  // Update an existing user
  const updateUser = async (id: string, data: UpdateUserRequest): Promise<UserResponse | null> => {
    setLoading(true);
    try {
      const response = await userApi.updateUser(id, data);
      return response.data || null;
    } catch {
      return null;
    } finally {
      setLoading(false);
    }
  };

  // Change user password
  const changePassword = async (id: string, data: ChangePasswordRequest): Promise<boolean> => {
    setLoading(true);
    try {
      await userApi.changePassword(id, data);
      return true;
    } catch {
      return false;
    } finally {
      setLoading(false);
    }
  };

  return {
    user,
    loading,
    fetchUserById,
    createUser,
    updateUser,
    changePassword,
    fetchUserByToken,
  };
};
