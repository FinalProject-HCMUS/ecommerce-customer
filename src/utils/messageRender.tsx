import { toast } from 'react-toastify';

const showManyError = (errors: string[]): void => {
  for (let i = 0; i < errors.length; i++) {
    toast.error(errors[i], {
      position: 'bottom-right',
    });
  }
};

const showError = (message: string | string[]): void => {
  if (Array.isArray(message)) {
    showManyError(message);
    return;
  }
  toast.error(message, {
    position: 'bottom-right',
  });
};

const showSuccess = (message: string) => {
  toast.success(message, {
    position: 'bottom-right',
  });
};

export { showManyError, showError, showSuccess };
