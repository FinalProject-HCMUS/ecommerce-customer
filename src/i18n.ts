import { initReactI18next } from 'react-i18next';
import i18n from 'i18next';

import localizationConstants from './constants/localization';
import { getCurrentLanguage } from './helpers/localization';

const { RESOURCES, REGIONS } = localizationConstants;

i18n.use(initReactI18next).init({
  resources: RESOURCES,
  lng: getCurrentLanguage(),
  fallbackLng: REGIONS.vi.key,

  interpolation: {
    escapeValue: false,
  },
});

export default i18n;
