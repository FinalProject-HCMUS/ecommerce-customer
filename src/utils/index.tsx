import * as formatDateUtils from '../utils/formatDate';
import * as messageRenderUtils from '../utils/messageRender';

export { formatDateUtils, messageRenderUtils };
