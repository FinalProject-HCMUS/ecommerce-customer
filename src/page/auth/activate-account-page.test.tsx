import React from 'react';
import { render } from '@testing-library/react';
import '@testing-library/jest-dom';

// Mock React hooks
jest.mock('react', () => {
  const originalReact = jest.requireActual('react');
  return {
    ...originalReact,
    useState: jest.fn(),
    FormEvent: jest.fn(),
    ChangeEvent: jest.fn(),
  };
});

// Restore React hooks implementation for each test
const mockSetState = jest.fn();
beforeEach(() => {
  // Mock useState to return values and setter functions
  (React.useState as jest.Mock).mockImplementation((initialValue) => [
    initialValue,
    mockSetState,
  ]);
});

// Mock the component under test
// import MockActivateAccountPage from '../__mocks__/activate-account-page';
const MockActivateAccountPage = () => (
  <div data-testid="mock-activate-account-page" />
);

jest.mock('./activate-account-page', () => {
  const ActivateAccountPage = (props: any) => {
    return <MockActivateAccountPage {...props} />;
  };
  return { __esModule: true, default: ActivateAccountPage };
});

// Mock hooks and utilities
jest.mock('../../hooks/user', () => ({
  useUser: jest.fn(() => ({
    resendUserConfirmationEmail: jest.fn(),
    loading: false,
  })),
}));

jest.mock('../../utils/messageRender', () => ({
  showError: jest.fn(),
  showSuccess: jest.fn(),
}));

jest.mock('../../helpers/i18n', () => ({
  t: jest.fn((key) => key),
}));

// Mock components
jest.mock('../../components/shared/form/InputField', () => ({
  __esModule: true,
  default: ({
    id,
    label,
    type,
    value,
    onChange,
    onBlur,
    placeholder,
    required,
  }: {
    id: string;
    label: string;
    type: string;
    value: string;
    onChange: React.ChangeEventHandler<HTMLInputElement>;
    onBlur: React.FocusEventHandler<HTMLInputElement>;
    placeholder?: string;
    required?: boolean;
  }) => (
    <div data-testid="input-field">
      <label htmlFor={id}>{label}</label>
      <input
        id={id}
        data-testid={id}
        type={type}
        value={value}
        onChange={onChange}
        onBlur={onBlur}
        placeholder={placeholder}
        required={required}
      />
    </div>
  ),
}));

jest.mock('../../components/shared/Button', () => ({
  GeneralButton: ({
    children,
    type,
    disabled,
    className,
    onClick,
  }: {
    children: React.ReactNode;
    type?: 'button' | 'submit' | 'reset';
    disabled?: boolean;
    className?: string;
    onClick?: React.MouseEventHandler<HTMLButtonElement>;
  }) => (
    <button
      type={type}
      disabled={disabled}
      className={className}
      onClick={onClick}
      data-testid="submit-button"
    >
      {children}
    </button>
  ),
}));

jest.mock('lucide-react', () => ({
  Mail: () => <svg data-testid="mail-icon" />,
}));

// Mock testing libraries
jest.mock('@testing-library/react', () => {
  const original = jest.requireActual('@testing-library/react');
  return {
    ...original,
    // Add any custom overrides here if needed
  };
});

jest.mock('@testing-library/jest-dom', () => {
  const original = jest.requireActual('@testing-library/jest-dom');
  return {
    ...original,
    // Add any custom overrides here if needed
  };
});

// Import after mocking
import ActivateAccountPage from './activate-account-page';
import { useUser } from '../../hooks/user';
import { showError, showSuccess } from '../../utils/messageRender';
import { t } from '../../helpers/i18n';

describe('ActivateAccountPage', () => {
  // Setup mock values and functions
  const mockResendUserConfirmationEmail = jest.fn<Promise<boolean>, [string]>();
  const mockShowError = jest.fn();
  const mockShowSuccess = jest.fn();

  beforeEach(() => {
    jest.clearAllMocks();

    // Setup mock implementations
    (useUser as jest.Mock).mockReturnValue({
      resendUserConfirmationEmail: mockResendUserConfirmationEmail,
      loading: false,
    });

    (showError as jest.Mock).mockImplementation(mockShowError);
    (showSuccess as jest.Mock).mockImplementation(mockShowSuccess);

    (t as jest.Mock).mockImplementation((key: string) => {
      const translations: Record<string, string> = {
        'lbl.activateAccount': 'Activate Account',
        'lbl.email': 'Email',
        'btn.sendConfirmationEmail': 'Send Confirmation Email',
        'error.emailRequired': 'Email is required',
        'success.sendEmailActivation': 'Activation email sent successfully',
        'success.sendEmailActivationFailed': 'Failed to send activation email',
        'lbl.checkEmail': 'Check your email',
        'btn.sendAnother': 'Send Another',
      };
      return translations[key] || key;
    });

    // Default mock implementation
    mockResendUserConfirmationEmail.mockImplementation(async () => true);
  });

  test('matches snapshot in initial state', () => {
    const { container } = render(<ActivateAccountPage />);
    expect(container).toMatchSnapshot('initial-state');
  });
});
