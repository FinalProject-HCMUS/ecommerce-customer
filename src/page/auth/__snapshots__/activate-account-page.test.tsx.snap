// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ActivateAccountPage matches snapshot in initial state: initial-state 1`] = `
<div>
  <div
    data-testid="mock-activate-account-page"
  />
</div>
`;
